lib/parse.js

```diff
@@ -46,14 +46,64 @@ const DEFAULT_VAR_PARSER = {
   image: {
     var: parseSelect,
     advanced: parseVarXStyle
-  }
+  },
+  number: parseRange,
+  range: parseRange
 };
 
 const DEFAULT_VAR_VALIDATOR = {
-  checkbox: validateCheckbox
+  checkbox: validateCheckbox,
+  number: validateRange,
+  range: validateRange
 };
 
 const MANDATORY_META = ['name', 'namespace', 'version'];
+const RANGE_PROPS = ['default', 'min', 'max', 'step'];
+
+function parseRange(state) {
+  parseJSON(state);
+  const result = {
+    min: null,
+    max: null,
+    step: null,
+    units: null
+  };
+  if (typeof state.value === 'number') {
+    result.default = state.value;
+  } else if (Array.isArray(state.value)) {
+    let i = 0;
+    for (const item of state.value) {
+      if (typeof item === 'string') {
+        if (result.units != null) {
+          throw new ParseError({
+            code: 'invalidRangeMultipleUnits',
+            message: 'units is alredy defined',
+            index: state.valueIndex
+          });
+        }
+        result.units = item;
+      } else if (typeof item === 'number' || item === null) {
+        if (i < RANGE_PROPS.length) {
+          result[RANGE_PROPS[i++]] = item;
+        }
+      } else {
+        throw new ParseError({
+          code: 'invalidRangeValue',
+          message: 'value must be number, string, or null',
+          index: state.valueIndex
+        });
+      }
+    }
+  } else {
+    throw new ParseError({
+      code: 'invalidRange',
+      message: 'the default value must be an array or a number',
+      index: state.valueIndex
+    });
+  }
```

Comment:
Should all these messages be put into a message object, so once we incorporate it into Stylus we can switch the locale easier?

Reply:
Since these errors have a `.code` property, we can put the translation in `messages.json` and translate the error with something similar to:
```js
try {
  // parse the meta
} catch (err) {
  if (err.code) {
    const message = chrome.getMessage(`metaError_${err.code}`, err.args);
    if (message) {
      err.message = message;
    }
    console.log(err.message);
  }
}
```